activerecord/lib/active_record/relation/predicate_builder.rb

```diff
@@ -112,6 +112,12 @@ def create_binds_for_hash(attributes)
               binds.concat(bvs)
               attrs
             end
+          when value.is_a?(Array) && !table.type(column_name).respond_to?(:subtype)
```

Comment:
I'd really prefer not to add more code which relies on type internals in this way. I think the code below this which does the same is a major smell. There's a greater problem with the structure here that we should address first.